### PR TITLE
feat(create-repository): add create repository action

### PR DESCRIPTION
**Pull request by vmaerten**
It's hard to create a new repository. You need to think to multiple things : 
- protect the main branch to enforce review
- create 3 env: dev, beta, prod
- protect the prod environment to ask for review deployment from the "dev" team
- allow only squash and merge in PR
- etc

This action does this for you. Is it used in the https://github.com/pbstck/pubstack/actions/workflows/create-repository.yaml as a manual workflow
I tested with https://github.com/pbstck/test-vmn

It uses https://github.com/pbstck/template as a template for the repository.
The goal is to provide a template with infra/, makefile, ci/cd in it.

### CODE REVIEW

File: create-repository/action.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: 'Create Repository'
+description: 'Greet someone and record the time'
```
Comment (slezier) [inline create-repository/action.yml:2]: 'Create repo compliant with team usage ' ?

Comment (vmaerten) [reply create-repository/action.yml:2]: I missed it 
I agree ! :)

**Review by slezier — APPROVED** [review]

**Review by ildede — APPROVED** [review]

File: create-repository/index.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    const [owner,] = process.env.GITHUB_REPOSITORY.split("/");
+    const name = core.getInput('name');
+    const description = core.getInput('description');
+    const allEnv = ['dev','beta','prod']
+    try {
+        const allTeam = await  octokit.rest.teams.list({
+            org: owner,
+        });
+        const devTeam = allTeam.data.find(d => d.name === 'dev')
+        console.log('Create repository using template  : ', name)
```
Comment (ildede) [inline create-repository/index.js:16]: This log tell me _"I'm creating the new repository using the template with name: xxx"_

I suggest something like:
```suggestion
        console.log(`Creating new reposityry ${name} using pbstck/template`)
```

Comment (vmaerten) [reply create-repository/index.js:16]: Thanks !

**Comment by tvincent2** [comment]
Doesn't it also need an update of README.md?
I see that most actions are described there.

**Comment by vmaerten** [comment]
> Doesn't it also need an update of README.md?
I see that most actions are described there.

You're right, I'll do it!

**Comment by vmaerten** [comment]
Done !

**Review by sdeluce — APPROVED** [review]

**Review by lestrade-pbstck — APPROVED** [review]

**Review by erwanncloarec — APPROVED** [review]

**Pull request merged**